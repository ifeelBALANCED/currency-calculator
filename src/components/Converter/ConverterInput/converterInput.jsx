import React from 'react';
import {Input} from "@chakra-ui/react";
const ConverterInput = ({amount, changeHandler}) => {
    return (
        <Input
            name="amount"
            type="text"
            mt={5}
            value={amount}
            onChange={changeHandler}
        />
    );
};

export default ConverterInput;
