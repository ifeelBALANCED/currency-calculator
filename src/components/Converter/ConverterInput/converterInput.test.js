import React from "react";
import ConverterInput from "./converterInput";

const setUpInput = (props) => shallow(<ConverterInput {...props}/>)
describe('should render ConverterInput component', () => {
    let component
    // initializing a component setup with setUpInput function
    beforeEach(() => {
        component = setUpInput()
    })
    describe('should test ConverterInput inner elements', () => {
        // expects that it would be 1 input in ConverterInput component
        it('should contain input', () => {
            const wrapper = component.find('input')
            expect(wrapper).toMatchSnapshot()
        });
        // expects render ConverterInput component with props
        it('should render ConverterInput component with props', () => {
            component = shallow(<ConverterInput amount={9} changeHandler={() => {
            }}/>)
            expect(component).toMatchSnapshot()
        });
        // expects render ConverterInput component without props
        it('should render ConverterInput component without props', () => {
            component = shallow(<ConverterInput/>)
            expect(component).toMatchSnapshot()
        });
    });
    // check ConverterInput handlers (onChange)
    describe('should test ConverterInput handlers', () => {
        it('should call onChange prop with input value', () => {
            const onChangeMock = jest.fn();
            component = mount(<ConverterInput onChange={onChangeMock} value="1"/>);
            const input = component.find('input');
            input.simulate('change');
            expect(onChangeMock.mock.calls.length).toEqual(0)
        });
    });
})
