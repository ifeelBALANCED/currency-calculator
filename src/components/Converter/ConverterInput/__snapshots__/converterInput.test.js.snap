// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConverterInput component should test ConverterInput inner elements should contain input 1`] = `null`;

exports[`should render ConverterInput component should test ConverterInput inner elements should render ConverterInput component with props 1`] = `
<Input
  mt={5}
  name="amount"
  onChange={[Function]}
  type="text"
  value={9}
/>
`;

exports[`should render ConverterInput component should test ConverterInput inner elements should render ConverterInput component without props 1`] = `
<Input
  mt={5}
  name="amount"
  type="text"
/>
`;
